polygon/heimdall/heimdall.go

```diff
@@ -579,7 +586,9 @@ func (h *heimdall) batchFetchCheckpoints(
 	sort.Sort(checkpoints)
 
 	for i, checkpoint := range checkpoints[lastStored:] {
-		err := store.PutCheckpoint(ctx, CheckpointId(i+1), checkpoint)
+		// checkpoint list API does not return "id" in the json response
+		checkpoint.Id = CheckpointId(i + 1)
+		err := store.PutCheckpoint(ctx, checkpoint.Id, checkpoint)
```

Comment:
IMO this logic should be a part of the HeimdallClient impl. Then it will apply to both service.go and heimdall.go

Reply:
@battlmonstr I thought about this however the `heimdall.HttpClient` does not know what `id` to assign to the `Checkpoint` in the current implementation of `FetchCheckpoints(page, limit)` because it can't infer that information from the `page` and `limit` only (due to the the bug heimdall has of not returning sorted items in the paging API).

I thought about introducing `FetchAllCheckpoint` method into `heimdall.HttpClient` which will work for `heimdall.Heimdall` but then it won't work for the `entityFetcherImpl` abstraction - it would need changes to the abstraction. Do you want me to create `FetchAllCheckpoint` and use that in `heimdall.Heimdall` for now and then let you incorporate that in the `entityFetcherImpl` you are working on?

As an aside probably the best solution would be for the paging API to return the "id" attribute in the json response. We can patch Heimdall by opening a PR and making a contribution. I've added this to our backlog of tasks. But until then we need to use this workaround.